### PR TITLE
feat(control_evaluator, tier4_control_launch): add a trigger to choice whether to output metrics to log folder

### PR DESCRIPTION
**Pull request by xtk8532704**
## Description
1. Add a bool parameter `output_metrics`, if it's true, the control_evaluator node writes the statics of the metrics measured during its lifetime to `<ros2_logging_directory>/autoware_metrics/<node_name>-<time_stamp>.json` when shut down.
This feature will be used to collect and analyze test results in real vehicle tests and evaluator tests.
2. Refactored control_evaluator's launch method to make it clearer: it now launches from `autoware_control_evaluator/launch/control_evaluator.launch.xml`, and can read `param/control_evaluator.defaults.yaml` correctly.

This feature will be used to collect and analyze test results in real vehicle tests and evaluator tests.

## Related links

**Parent Issue:**

- Link

## How was this PR tested?
Psim with `output_metrics=true`:
```bash
$ cat /home/temkeikem/.ros/log/autoware_metrics/autoware_control_evaluator-2024-11-27-00-28-32.json 
{
    "goal_lateral_deviation/count": 0,
    "goal_lateral_deviation/description": "Lateral deviation from the goal point[m]",
    "goal_lateral_deviation/max": -1.7976931348623157e+308,
    "goal_lateral_deviation/mean": 0.0,
    "goal_lateral_deviation/min": 1.7976931348623157e+308,
    # ignore many outputs.
    "yaw_deviation/count": 227,
    "yaw_deviation/description": "Yaw deviation from the reference trajectory[rad]",
    "yaw_deviation/max": 0.06566750430257973,
    "yaw_deviation/mean": 0.031098377450999377,
    "yaw_deviation/min": 4.124327548060691e-05
}
```

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp [hunk 1]
```diff
@@ -85,9 +91,18 @@ class ControlEvaluatorNode : public rclcpp::Node
...
   void getRouteData();
 
-  // Calculator
-  // Metrics
-  std::deque<rclcpp::Time> stamps_;
+  // Parameters
+  bool output_metrics_;
+
+  // Metric
+  std::vector<Metric> metrics_ = {
```
Comment (kosuke55) [inline evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp:98]:
```suggestion
  const std::vector<Metric> metrics_ = {
```

File: evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp [hunk 2]
```diff
@@ -85,9 +91,18 @@ class ControlEvaluatorNode : public rclcpp::Node
...
+    // collect all metrics
+    Metric::lateral_deviation,      Metric::yaw_deviation,      Metric::goal_longitudinal_deviation,
+    Metric::goal_lateral_deviation, Metric::goal_yaw_deviation,
+  };
+
+  std::array<Accumulator<double>, static_cast<size_t>(Metric::SIZE)>
+    metric_accumulators_;  // 3(min, max, mean) * metric_size
```
Comment (kosuke55) [inline evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp:105]: (nits: you can use unorder map?)

File: evaluator/autoware_control_evaluator/src/control_evaluator_node.cpp [hunk 3]
```diff
@@ -33,12 +37,63 @@ ControlEvaluatorNode::ControlEvaluatorNode(const rclcpp::NodeOptions & node_opti
...
+  using json = nlohmann::json;
+  json j;
+  for (Metric metric : metrics_) {
+    const std::string base_name = metric_to_str.at(metric) + "/";
+    j[base_name + "min"] = metric_accumulators_[static_cast<size_t>(metric)].min();
+    j[base_name + "max"] = metric_accumulators_[static_cast<size_t>(metric)].max();
+    j[base_name + "mean"] = metric_accumulators_[static_cast<size_t>(metric)].mean();
+    j[base_name + "count"] = metric_accumulators_[static_cast<size_t>(metric)].count();
+    j[base_name + "description"] = metric_descriptions.at(metric);
+  }
```
Comment (kosuke55) [inline evaluator/autoware_control_evaluator/src/control_evaluator_node.cpp:65]: if writing like
```
for (size_t i = 0; i < static_cast<sizse_t>(Metric::SIZE); ++i) {
j[base_name + "min"] = metric_accumulators_[i].min();
```
metrics_ variable is not necessary?

File: evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp (hunk 1 above)
Comment (kosuke55) [reply evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp:98]: (this member may not necessary)

File: evaluator/autoware_control_evaluator/src/control_evaluator_node.cpp (hunk 3 above)
Comment (xtk8532704) [reply evaluator/autoware_control_evaluator/src/control_evaluator_node.cpp:65]: I want to get `metric_descriptions`:
`j[base_name + "description"] = metric_descriptions.at(metric);`

File: evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp (hunk 1 above)
Comment (kosuke55) [reply evaluator/autoware_control_evaluator/include/autoware/control_evaluator/control_evaluator_node.hpp:98]: this is for set on/off metrics in the future.

**Review by kosuke55 — APPROVED** [review]

**Review by danielsanchezaran — APPROVED** [review]
LGTM

**Pull request merged**